python/lsst/pipe/tasks/matchBackgrounds.py

```diff
@@ -19,683 +19,627 @@
-                               (len(expRefList), len(imageScalerList)))
-
-        for expRef, imageScaler in zip(expRefList, imageScalerList):
-            exposure = expRef.get()
-            maskedImage = exposure.getMaskedImage()
-            if imageScaler is not None:
-                try:
-                    imageScaler.scaleMaskedImage(maskedImage)
-                except Exception:
-                    # need to put a place holder in Arr
-                    varList.append(numpy.nan)
-                    meanBkgdLevelList.append(numpy.nan)
-                    coverageList.append(numpy.nan)
-                    continue
-            statObjIm = afwMath.makeStatistics(maskedImage.getImage(), maskedImage.getMask(),
-                                               afwMath.MEAN | afwMath.NPOINT | afwMath.VARIANCE, self.sctrl)
-            meanVar, meanVarErr = statObjIm.getResult(afwMath.VARIANCE)
-            meanBkgdLevel, meanBkgdLevelErr = statObjIm.getResult(afwMath.MEAN)
-            npoints, npointsErr = statObjIm.getResult(afwMath.NPOINT)
-            varList.append(meanVar)
-            meanBkgdLevelList.append(meanBkgdLevel)
-            coverageList.append(npoints)
-        if not coverageList:
-            raise pipeBase.TaskError(
-                "None of the candidate %s exist; cannot select best reference exposure" % (expDatasetType,))
-
-        # Normalize metrics to range from  0 to 1
-        varArr = numpy.array(varList)/numpy.nanmax(varList)
-        meanBkgdLevelArr = numpy.array(meanBkgdLevelList)/numpy.nanmax(meanBkgdLevelList)
-        coverageArr = numpy.nanmin(coverageList)/numpy.array(coverageList)
-
-        costFunctionArr = self.config.bestRefWeightVariance * varArr
-        costFunctionArr += self.config.bestRefWeightLevel * meanBkgdLevelArr
-        costFunctionArr += self.config.bestRefWeightCoverage * coverageArr
-        return numpy.nanargmin(costFunctionArr)
-
-    @timeMethod
-    def matchBackgrounds(self, refExposure, sciExposure):
-        """Match science exposure's background level to that of reference exposure.
+        # Extract mean/var/npoints for each visit background model
+        fitChi2s = []  # Background goodness of fit
+        fitVars = []  # Variance
+        fitNPointsGlobal = []  # Global coverage
+        visits = []  # To ensure dictionary key order is correct
+        for vis in visitTractBackgrounds:
+            visits.append(vis)
+            # Fit a polynomial model to the full tract plane
+            tractBg = visitTractBackgrounds[vis].getStatsImage()
+            fitBg, _ = self._makeBackground(tractBg, binSize=1)
+            # TODO: as stated above, fitting a pre-binned image results in a
+            # null variance image.  But we want to add variance into the cost
+            # function.  How best to do that?  Below is a bad temporary
+            # solution, just assuming variance = mean
```

Comment:
I'd recommend computing an estimate of the variance earlier and attaching it to the tract backgrounds (i.e. as a scalar).  `variance = mean` is definitely not going to hold in nJy.